coral/src/app/features/connectors/request/ConnectorPromotionRequest.tsx

```diff
@@ -0,0 +1,364 @@
+                    borderWidth={1}
+                    paddingY={"3"}
+                    borderRadius={2}
+                  >
+                    <MonacoEditor
+                      data-testid="connector-config"
+                      height="200px"
+                      language="json"
+                      theme={"light"}
+                      onChange={(value) => {
+                        value !== undefined &&
+                          form.setValue("connectorConfig", value, {
+                            shouldValidate: true,
+                          });
+                      }}
+                      options={{
+                        language: "json",
+                        ariaLabel: "Connector configuration, required",
+                        renderControlCharacters: false,
+                        minimap: { enabled: false },
+                        folding: false,
+                        lineNumbers: "off",
+                        scrollBeyondLastLine: false,
+                        renderLineHighlight: "none",
+                        cursorBlinking: "solid",
+                        overviewRulerLanes: 0,
+                        wordBasedSuggestions: false,
+                        lineNumbersMinChars: 3,
+                        glyphMargin: false,
+                        cursorStyle: "line-thin",
+                        scrollbar: {
+                          useShadows: false,
+                          verticalScrollbarSize: 2,
+                        },
+                        fixedOverflowWidgets: true,
+                      }}
+                      value={field.value}
+                    />
+                  </BorderBox>
+                  <Box marginTop={"1"} marginBottom={"3"} aria-hidden={"true"}>
+                    <Typography.Caption color={"error-50"}>
+                      {error !== undefined ? error.message : <>&nbsp;</>}
+                    </Typography.Caption>
+                  </Box>
+                </>
+              );
+            }}
+          />
+
+          <Grid colGap={"l1"} marginTop={"3"} cols={"2"}>
+            <Textarea<ConnectorRequestFormSchema>
+              name={"description"}
+              value={connectorDetails?.connectorContents?.description}
+              labelText={"Connector description (read-only)"}
+              readOnly={true}
+            />
+            <Textarea<ConnectorRequestFormSchema>
+              name={"remarks"}
+              labelText={"Message for approval"}
+            />
+          </Grid>
+
+          <Box display={"flex"} colGap={"l1"} marginTop={"3"}>
+            <SubmitButton loading={isLoadingRequestPromotion}>
+              Submit promotion request
+            </SubmitButton>
+            <Button
+              type="button"
+              kind={"secondary"}
+              disabled={isLoadingRequestPromotion}
+              onClick={() => {
+                if (!isEmpty(form.formState.touchedFields)) {
```

Comment:
Yes! `isDirty` will always be `true`, since the fixed values are set by `values`, which is reactive, which means they are updating once e.g. `connectorDetails` is fetched.

Reply:
Ah yes! I think this is why this is why I chose to use `form.reset`  for the Topic promotion form, which creates new initial values after the necessary data has been fetched, so  we can have the form behaving a little bit more idiomatically with `initialValues` once the data has been fetched. https://github.com/Aiven-Open/klaw/blob/main/coral/src/app/features/topics/request/TopicPromotionRequest.tsx#L134-L144

Not sure which approach is preferable, but probably we'll find out once the forms get a little bit of use ^^

Reply:
Yes, I think so, too!